### PR TITLE
Feature Signin

### PR DESCRIPTION
**Pull request by lsh9034**
Implement Login session in Apollo server by jwt and context function to check whether the query have qualification.

### CODE REVIEW

File: web/src/components/login-form.js [hunk 1]
```diff
@@ -1,16 +1,99 @@
...
+
+  // After authenticated from server, redirect even if success or not.
+  redirect = () => {
+    if(this.state.isMember)
+      return this.props.history.push('/');
+    else
+      return this.props.history.push('/signup');
+  };
+
+  // social login button and send googleId received from google to server using mutation component
```
Comment (ethkim) [inline web/src/components/login-form.js]: Break into multiple lines if lines exceed 80 chars. Proper capitalization.

File: web/src/components/login-form.js [hunk 2]
```diff
@@ -1,16 +1,99 @@
...
   render() {
     return (
         <div>
-
-        </div>
-    );
+          <Mutation mutation={SIGNIN_QUERY} variables={{googleId: this.state.googleId}}
+              onCompleted={
+                (data)=>{
+                  this.changeIsMember(/* isSucess= */ data.signInWithGoogle.success, /* token= */ data.signInWithGoogle.token);
```
Comment (ethkim) [inline web/src/components/login-form.js]: Break line.

File: web/src/pages/signin.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import React, {Component} from 'react';
+import LoginForm from '../components/login-form'
+
+//TODO(Hyejin): implement signin page using material UI
```
Comment (ethkim) [inline web/src/pages/signin.js]: Space. capitalization.

File: web/src/components/login-form.js [hunk 4]
```diff
@@ -1,16 +1,99 @@
...
+import {clientId} from './utils.js';
+import {gql} from 'apollo-boost';
+import {Mutation} from 'react-apollo'
+import {withRouter} from 'react-router-dom'
+
+/*query sending user Information to server*/
+const SIGNIN_QUERY = gql`
+        mutation ($googleId: String!){
+          signInWithGoogle(googleId: $googleId){
+            success
```
[the remark this answers is not in the document]
Comment (lsh9034) [reply web/src/components/login-form.js]: commas doesn't work in graphql query.

File: server/src/resolvers/mutation.js [hunk 5]
```diff
@@ -15,13 +12,13 @@ module.exports.Mutation = {
       };
     }
 
-    // TODO(lsh9034): Implement session logic.
-    // Create session and return sessionId to client.
+    // TODO(lsh9034): modify expires time depend on last user interaction.
```
Comment (ethkim) [inline server/src/resolvers/mutation.js]: "Modify expiration time depending on the last user interaction."

File: server/src/resolvers/mutation.js [hunk 6]
```diff
@@ -15,13 +12,13 @@ module.exports.Mutation = {
...
+    const token = jwt.sign({userId:userId}, APP_SECRET,{expiresIn:'100h'});
```
Comment (ethkim) [inline server/src/resolvers/mutation.js]: Token expiration seems to be set to different time duration between be and fe (or at least they're declared in multiple different places). Can we find ways to unify them?

**Pull request merged**